test/integration/selenium-helpers.js

```diff
@@ -146,23 +250,30 @@ class SeleniumHelper {
     /**
      * Find an element by xpath.
      * @param {string} xpath The xpath to search for.
-     * @param {string} timeoutMessage The message to use if the element is not found.
-     * A different (non-configurable) message will be used if the element is found but not visible.
      * @returns {Promise<webdriver.WebElement>} A promise that resolves to the element.
      */
-    async findByXpath (xpath, timeoutMessage = `findByXpath timed out for path: ${xpath}`) {
-        const el =
-            await this.driver.wait(until.elementLocated(By.xpath(xpath)), DEFAULT_TIMEOUT_MILLISECONDS, timeoutMessage);
-        await this.driver.wait(el.isDisplayed(), DEFAULT_TIMEOUT_MILLISECONDS, `${xpath} is not visible`);
-        return el;
+    async findByXpath (xpath) {
+        const outerError = new SeleniumHelperError('findByXpath failed', [{xpath}]);
```

Comment:
The main job of this `new` is to capture the stack in a way that will include the containing function. As a side bonus, it builds the `message` property in a convenient way.